### PR TITLE
tiny fix iterative checkpoint test case

### PR DESCRIPTION
**Pull request by liusdu**
Criu creates symlink named "parent" under image-path while doing
interactive checkpoint, without checking wether symlink points to
right place. This patch corrects related test case.

Signed-off-by: Liu Hua <weldonliu@tencent.com>

### CODE REVIEW

**Comment by kolyshkin** [comment]
CI got stuck in Fedora -- restarted.

**Comment by kolyshkin** [comment]
@adrianreber PTAL 🙏🏻

**Comment by adrianreber** [comment]
The change looks correct. `--parent-path` needs to point to the previous checkpoint directory from the current checkpoint's directory point of view.

@liusdu have you seen any errors? Where did you seen errors. Just curious.

**Comment by liusdu** [comment]
> The change looks correct. `--parent-path` needs to point to the previous checkpoint directory from the current checkpoint's directory point of view.
> 
> @liusdu have you seen any errors? Where did you seen errors. Just curious.

 @kolyshkin  I  repushed  the commit. And all tests passed.

@adrianreber  I did iterative checkpoint test and got several types of error  with wrong `--parent-path` value:
1.  invalid `--parent-path` : Last dump has everything and pre-dump is useless at all ( what runc test does)
2. right `--parent-path`  but with absolute path: all dumps go well but restore will hang(criu just ignore partent images)

So I went into runc code to debug. And I found this wrong test.

Bte, should I add some piece of code to validate  this?

**Comment by liusdu** [comment]
@adrianreber @calavera  I add a patch to validate parent path. 
Please take a look. I can revert it if not necessary.

File: checkpoint.go
Comment (kolyshkin) [inline checkpoint.go:84]: no need to add a newline here

File: checkpoint.go
Comment (kolyshkin) [inline checkpoint.go:94]: Please remove this newline

File: checkpoint.go
Comment (kolyshkin) [inline checkpoint.go:99]: I guess you should check first if `parentPath` is absolute, and error out if it's not
```go
if !filepath.IsAbs(parentPath) {
   return errors.New("--parent-path must be relative")
}
```
or so.

File: tests/integration/checkpoint.bats
Comment (kolyshkin) [inline tests/integration/checkpoint.bats:173]: So what I still don't understand is why the test was not failing.

File: checkpoint.go
Comment (kolyshkin) [inline checkpoint.go:101]: I think this belongs to option description, in here you can just return `fmt.Errorf("invalid --parent-path %s: %w", parentPath, err)` or something like that.

Comment (kolyshkin) [reply checkpoint.go:101]: ...with
```go
if err == nil && !fi.IsDir {
   err := &os.PathError(Path: realParent, Err: unix.ENOTDIR)
}
```

File: tests/integration/checkpoint.bats
Comment (kolyshkin) [inline tests/integration/checkpoint.bats]: simple_cr() is called from many places, I think this deserves a separate test case. Also, I guess it's enough to check that runc fails, and the error contains "--parent-path", rather than checking for a full specific error message (which might change).

File: tests/integration/checkpoint.bats
Comment (adrianreber) [reply tests/integration/checkpoint.bats:173]: I think that is CRIU's fault and confuses many people when using pre-dump. CRIU silently switched to non pre-dump if `--parent-path` is not correct. https://github.com/checkpoint-restore/criu/pull/1403

Restore also works without any message, but it is not using the pre-dump. This will probably be an error in the next CRIU release. Having it fixed in runc already know makes sense.

File: tests/integration/checkpoint.bats
Comment (liusdu) [reply tests/integration/checkpoint.bats]: @kolyshkin Thanks very much for you review messages～.  I add a test case  to  test  `--parent-path`. Am I missing something？

File: tests/integration/checkpoint.bats
Comment (kolyshkin) [inline tests/integration/checkpoint.bats:148]: extra empty line

File: tests/integration/checkpoint.bats
Comment (kolyshkin) [inline tests/integration/checkpoint.bats:150]: As you don't really checkpoint or restore anything, there's no need to do `setup_pipes` / `runc_run_with_pipes` / `check_pipes`. A simple
```bats
        runc run -d --console-socket "$CONSOLE_SOCKET" test_busybox
        [ "$status" -eq 0 ]
```
should be enough.

File: tests/integration/checkpoint.bats
Comment (kolyshkin) [inline tests/integration/checkpoint.bats:173]: As per the above comment, remove this. This is to check that the restored container works, we don't restore the container so ...

File: tests/integration/checkpoint.bats
Comment (kolyshkin) [inline tests/integration/checkpoint.bats:159]: this comment is misleading, please remove

**Comment by kolyshkin** [comment]
A few comments about simplification of "bad --parent-path" test case. Otherwise LGTM

**Review by kolyshkin — APPROVED** [review]
LGTM

**Comment by liusdu** [comment]
ping @adrianreber

**Comment by liusdu** [comment]
@kolyshkin all test passed

**Comment by kolyshkin** [comment]
@liusdu I have approved these changes a while ago, waiting for the second LGTM. @opencontainers/runc-maintainers PTAL

**Comment by haircommander** [comment]
LGTM

**Pull request merged**
